packages/postcss-plugin/src/bundler.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ */
+
+const babel = require('@babel/core');
+const stylexBabelPlugin = require('@stylexjs/babel-plugin');
+
+// Creates a stateful bundler for processing StyleX rules using Babel.
+module.exports = function createBundler() {
+  const styleXRulesMap = new Map();
+
+  // Determines if the source code should be transformed based on the presence of StyleX imports.
+  function shouldTransform(sourceCode) {
+    return sourceCode.includes('stylex');
+  }
+
+  // Transforms the source code using Babel, extracting StyleX rules and storing them.
+  async function transform(id, sourceCode, babelConfig, options) {
```

Comment:
I think we could share some code between the postcss plugin and the CLI. Although this plugin should make the CLI redundant in most cases.

Reply:
Yes. The goal of this PR is to make the plugin simple, reliable and production-ready for Next.js and other bundler users so I made the plugin simple and self-contained.

It's out of the scope of this PR because of that but I have some success in making this work for Next.js + Turbopack setup as well ([gist](https://gist.github.com/javascripter/17f0fe9cc54cca0087c273aecc4412a6)). It depends on some internals and am not using it in my production setup but I see a path where StyleX CLI becomes unnecessary for Turbopack users too so it's worth decoupling this plugin from the CLI.

Reply:
I just mean we can refactor the CLI to share some of the code between it and the postcss plugin. Anyway, definitely out of scope for this PR.

Reply:
I think @javascripter's point is that it's simpler keeping the packages decoupled. There's no real value to sharing code here